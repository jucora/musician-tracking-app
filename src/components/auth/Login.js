/* eslint no-console: ["error", { allow: ["warn", "error"] }] */

import React from 'react';
import axios from 'axios';
import PropType from 'prop-types';

export default class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      // loginErrors: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    const { handleSuccessfulAuth } = this.props;
    const { email, password } = this.state;

    e.preventDefault();
    axios
      .post(
        'http://localhost:3001/sessions',
        {
          user: {
            email,
            password,
          },
        },
        { withCredentials: true }
      )
      .then((response) => {
        if (response.data.logged_in) {
          handleSuccessfulAuth(response.data);
        }
      })
      .catch((error) => {
        console.error('Login error', error);
      });
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { email, password } = this.state;
    return (
      <div className="form">
        <form onSubmit={this.handleSubmit}>
          <input
            type="email"
            name="email"
            placeholder="Your Email"
            value={email}
            onChange={this.handleChange}
            required
          />

          <input
            type="password"
            name="password"
            placeholder="Password"
            value={password}
            onChange={this.handleChange}
            required
          />

          <button type="submit">Login</button>
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  handleSuccessfulAuth: PropType.func.isRequired,
};
