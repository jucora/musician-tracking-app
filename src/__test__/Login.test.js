import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Login from '../components/auth/Login';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<Login handleSuccessfulAuth={() => ''} />);
  expect(asFragment()).toMatchSnapshot();
});
