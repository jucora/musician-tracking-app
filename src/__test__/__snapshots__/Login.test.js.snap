// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <form>
      <h2>
        Log In
      </h2>
      <h2
        class="error"
      />
      <input
        name="email"
        placeholder="Your Email"
        required=""
        type="email"
        value=""
      />
      <input
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
